PhotoStreamer/app/src/main/java/com/mtruehle/photostreamer/FeedFragment.java

```diff
@@ -0,0 +1,117 @@
+package com.mtruehle.photostreamer;
+
+import android.support.v4.app.Fragment;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebView;
+import android.widget.Button;
+import android.widget.RelativeLayout;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class FeedFragment extends Fragment {
+    private FeedDbHelper helper;
+    private WebView feedView;
+    private int currentlyDisplayedImage;
+    private ArrayList<String> savedUrls;
+
+    public FeedFragment() {
+        // Required empty public constructor
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        RelativeLayout feedLayout = (RelativeLayout) inflater.inflate(R.layout.fragment_feed, container, false);
+        Button backButton = (Button) feedLayout.findViewById(R.id.back_button_feed);
+        Button unsaveButton = (Button) feedLayout.findViewById(R.id.unsave_button_feed);
+        Button nextButton = (Button) feedLayout.findViewById(R.id.next_button_feed);
+        feedView = (WebView) feedLayout.findViewById(R.id.feed_webView);
+        helper = new FeedDbHelper(getContext());
+        updateSavedUrls();
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goToPrevious();
+            }
+        });
+        unsaveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                unsaveCurrent();
+            }
+        });
+        nextButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goToNext();
+            }
+        });
+        return feedLayout;
+
+    }
+
+    public void updateSavedUrls() {
+        savedUrls = helper.readFeedDb();
+        if (savedUrls.size() == 0) {
+            currentlyDisplayedImage = -1;
+        } else {
+            currentlyDisplayedImage = 0;
+            updateWebView();
+        }
+    }
+
+    public void updateWebView() {
+        String urlToShow = savedUrls.get(currentlyDisplayedImage);
+        try {
+            feedView.loadUrl(urlToShow);
+            Log.i("PRINTER", urlToShow);
+        } catch (Exception ex) {
+            ex.printStackTrace();
```

Comment:
Maybe this should return a boolean indicating success or failure so you can have the rest of code respond to the failure. Maybe by making a toast perhaps